src/main/webapp/ChapMap.js

```diff
@@ -475,4 +377,140 @@ class ChapMap {
 
     connection.send(JSON.stringify(jsonObject));
   }
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// MAP SHARING
+
+  /**
+   * @Private
+   * Sets the events related to sharing the map with another user
+   */
+  setMapShareEvents_() {
+    let shareFn = () => this.openSharePopup_();
+    this.addClickEvent_("shareBtnWrapper", shareFn);
+
+    let addEmailFn = () => this.addEmail_();
+    this.addClickEvent_("addEmail", addEmailFn);
+
+    let submitFn = () => this.submitSharing_();
+    this.addClickEvent_("share", submitFn);
+
+    let closeFn = () => this.closeSharePopup_();
+    this.addClickEvent_("close", closeFn);
+  }
+
+
+  /**
+   * @Private
+   * Sets a click-trigger event to the DOM element with the given id and
+   * sets the callback function to the one given
+   * @param {String} id the id of the element to be added
+   * @param {} fn the anonymous function to be called on click
+   */
+  addClickEvent_(id, fn) {
+    let btn = this.getEl_(id);
+    btn.addEventListener('click', fn);
+  }
+
+  /**
+   * @Private
+   * Opens the sharing popup and prevents the client from clicking on the map
+   */
+  openSharePopup_() {
+    let overlay = this.getEl_("share-popup");
+    overlay.classList.add("cover");
+
+    // let popup = this.getEl_("popup-window");
+    // popup.style.display="block";
+  }
+
+  /**
+   * @Private
+   * Closes the pop and allows clients to click on the map again
+   */
+  closeSharePopup_() {
+    let overlay = this.getEl_("share-popup");
+    overlay.classList.remove("cover");
+
+    // let popup = this.getEl_("popup-window");
+    // popup.style.display="none";
+
+    this.clearPopupInput_();
+  }
+
+  /**
+   * @Private
+   * Adds the given email to the email bank
+   */
+  addEmail_() {
+    let input = this.getEl_("email");
+    let emailDiv = this.createEmailDiv_(input.value);
+    input.value="";
+    let emailBank = this.getEl_("email-bank");
+    emailBank.appendChild(emailDiv);
+  }
+
+  /**
+   * @Private
+   * Creates a DOM element with the email and a delete button and adds it
+   * to the email bank
+   */
+  createEmailDiv_(email) {
+    let emailWrapper = document.createElement("div");
+    emailWrapper.classList.add("emailWrapper");
+    emailWrapper.id = email;
```

Comment:
id should only be used for styling, selecting, or anchoring. For something like this, it's recommended to use data attributes. These are custom attributes that start with the prefix `data-`. For example `data-email`.

Reply:
That's super useful! I'll change the code asap.